src/Idris/Elab/Implementation.idr

```diff
@@ -466,8 +466,10 @@ elabImplementation {vars} ifc vis opts_in pass env nest is cons iname ps named i
 
     mkTopMethDecl : (Name, Name, List (String, String), RigCount, Maybe TotalReq, RawImp) -> ImpDecl
     mkTopMethDecl (mn, n, upds, c, treq, mty)
-        = let opts = maybe opts_in (\t => Totality t :: opts_in) treq in
-              IClaim (MkFCVal vfc (MkIClaimData c vis opts (MkImpTy EmptyFC (NoFC n) mty)))
+        = do let opts = if isJust $ findTotality opts_in
+                          then opts_in
```

Comment:
Is this throwing away the explicit totality passed to this function quietly? Seems like that could result in surprising behavior.

Reply:
`treq` is a totality from a function declaration in the interface. If the interface implementation contains its own totality modifier, it is overrided. See https://github.com/idris-lang/Idris2/issues/3437#issuecomment-2529756651

Reply:
I think the only remaining question I have is whether the old behavior of having two modifiers attached (as you say, not useful) resulted in the same behavior of your new PR or if you are getting the _other_ modifier after your PR gets merged. I guess put another way, are you keeping the same modifier as would have "won" in the eyes of the totality checker before your change?

Reply:
I guess the answer to my question just be yes based on the examples you bring up in your comment on the linked issue, but want to make sure that couldn't be false.

Reply:
Yes, I'm keeping the modifier that would have won before changes. The implementation modifier is a higher priority now.